scripts/functional-tests.sh

```diff
@@ -132,6 +134,29 @@ wait_for_server(){
     fi
 }
 
+# It looks like GitHub action MacOS runner does not have graphics
```

Comment:
What was the reason for this change?

Reply:
I made the check into its own function given that the `cleanup()` overwrites the config file now, this avoids the back-and-forth `sed` (to revert) within functions and inside the cleanup. Since we now apply this change again during cleanup.

Reply:
Thanks!